cypress/robots/BaseRobot.ts

```diff
@@ -0,0 +1,232 @@
+    dom: string,
+    childDomClass: string,
+    minimumLength: number
+  ) {
+    cy.get(dom)
+      .find(`.${childDomClass}`)
+      .should("have.length.greaterThan", minimumLength);
+    return this;
+  }
+
+  seesNumberOfElementsInDom(
+    dom: string,
+    childDomClass: string,
+    length: number
+  ) {
+    cy.get(dom).find(`.${childDomClass}`).should("have.length", length);
+    return this;
+  }
+
+  seesPathNameInUrl(path: string) {
+    cy.location("pathname").should("eq", path);
+    return this;
+  }
+
+  seesFullPathNameWithQueryParams(path: string) {
+    const HOST = Cypress.env("APP_URL");
+    cy.location("href").should("eq", `${HOST}/${path}`);
+    return this;
+  }
+
+  seesDomDisabled(dom: string) {
+    cy.get(dom).should("be.disabled");
+    return this;
+  }
+
+  seesDomEnabled(dom: string) {
+    cy.get(dom).should("not.be.disabled");
+    return this;
+  }
+}
+
+export class BaseHands {
+  clickOnId(id: string) {
+    cy.get(`#${id}`).click();
+    return this;
+  }
+
+  doubleClickOnId(id: string) {
+    cy.get(`#${id}`).dblclick();
+    return this;
+  }
+
+  doubleClickAndDragOnAgGrid(rowId: string, colId: string) {
+    cy.get(`[row-id=${rowId}]`)
+      .find(`[col-id=${colId}]`)
+      .trigger("mousedown", { which: 1, pageX: 600, pageY: 100 })
+      .trigger("mousemove", { which: 1, pageX: 600, pageY: 600 })
+      .trigger("mouseup", { which: 1, pageX: 600, pageY: 6000 });
+    return this;
+  }
+
+  clickOnDomElement(dom: string) {
+    cy.get(dom).click();
+    return this;
+  }
+
+  clickOnDomElementText(text: string) {
+    cy.contains(text).click();
+  }
+
+  typeTextonDom(locatorName: string, locatorValue: string, text: string) {
+    cy.get(`[${locatorName}="${locatorValue}"]`).type(text, { force: true });
+    return this;
+  }
+
+  typeTextonId(id: string, text: string) {
+    cy.get(`#${id}`).type(text, { force: true });
+    return this;
+  }
+
+  clickOnChildDom(parentId: string, dom: string, index: number) {
+    cy.get(`#${parentId} ${dom}`).eq(index).click();
+    return this;
+  }
+
+  ClickOnTextWithClassAndIndex(domClass: string, index: number) {
+    cy.get(`[class=${domClass}]`).eq(index).click();
+    return this;
+  }
+
+  clickOnImageByPath(path: string) {
+    cy.getImgsBySrcPattern(path).click();
+  }
+
+  scrollToWithClassName(domClass: string, direction: PositionType) {
+    cy.get(`.${domClass}`).scrollTo(direction);
+    return this;
+  }
+
+  clickOnAgGridRow(rowId: string) {
+    cy.get(`[row-id=${rowId}]`).find(".ag-selection-checkbox").click();
+    return this;
+  }
+
+  wait(milliSecs: number) {
+    cy.wait(milliSecs);
+    return this;
+  }
+}
+
+export class BaseDependencies {
+  visitUrl(url: string) {
+    const HOST = Cypress.env("APP_URL");
+    cy.visit(`${HOST}${url}`);
+    return this;
+  }
+
+  accessUrl(url: string) {
+    cy.visit(`${url}`);
+    return this;
+  }
+
+  login() {
+    cy.get("#email").type(Cypress.env("USER_NAME"));
```

Comment:
check this

Reply:
it is a library function, written by zemoso.